### DISCUSSION TITLE
Viomi SE Support for Working Mode - Vaccum, Vaccum&Mop, Mop

### DISCUSSION THREAD

**Discussion by justdummyuser** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#465] [Q&A] [answered]
Hello Everyone,

Newbie on home automation world who is still learning how these things works. 

Recently i purchased Viomi SE Vacuum + Mop cleaner which I integrated in HA using marotoweb/viomise platform and PiotrMachowski/lovelace-xiaomi-vacuum-map-card.

Things are working good except below functions related to mopping feature -

1. From Mi Home app, this vacuum model support 3 work modes - "Vacuum-Only", "Vacuum&Mop" and "Mop-Only". But I couldn't figure out how to make them available in HA PiotrMachowski/lovelace-xiaomi-vacuum-map-card. By default it starts on "Vacuum-Only" mode.
2. I'm also looking for water level (low, medium, high ) setting while using "Vacuum&Mop" and "Mop-Only"

Any help is appreciated.

Thanks!!

**Comment by PiotrMachowski** [comment, marked as answer]
I think you should ask this question in marotoweb/viomise repo. I need to know appropriate service calls to add these features to the card.

**Reply by justdummyuser** [reply]
Alright. Thanks!

**Comment by gmp91** [comment]
Same for my Roborock S7 MaxV. 
So when I understand correclty, there is no service call to do the mopping? Only the devs of the robots API (correct?) can integrate this?
(sorry, newbie)

**Reply by PiotrMachowski** [reply]
Your vacuum uses a different integration then one mentioned in the discussion above. You should be able to enable mopping by changing fan speed to "Gentle"
